### PR TITLE
Slack: Add first pass slack integration with mlops events

### PR DESCRIPTION
**Pull request by sriram109**

### CODE REVIEW

**Review by sindhu-ghanta — APPROVED** [review]

File: components/Python/PySpark/slack-events/component.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  "program": "main.py",
+  "modelBehavior": "Auxiliary",
+  "useMLOps": true,
+  "inputInfo": [],
+  "outputInfo": [],
+  "group": "Algorithms",
+  "arguments": [
+    {
+      "key": "token",
+      "label": "slack token",
```
Comment (liororama) [inline components/Python/PySpark/slack-events/component.json]: "Slack Token"

File: components/Python/PySpark/slack-events/component.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      "type": "str",
+      "description": "slack app token",
```
Comment (liororama) [inline components/Python/PySpark/slack-events/component.json]: Slack application token

File: components/Python/PySpark/slack-events/main.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+    # Getting the first agent of ion component "0"
+    agent_list = pm.get_agents(pm.get_current_node().name)
+    if len(agent_list) == 0:
+        print("Error - must have agents this ion component is running on")
+        raise Exception("Agent list is empty")
+    return agent_list
+
+def get_channel_id(sc, name):
+    #ret = sc.api_call("channels.list")
```
Comment (liororama) [inline components/Python/PySpark/slack-events/main.py]: Remove commented code

File: components/Python/PySpark/slack-events/main.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        for ch in ret['channels']:
+            if ch['name'] == 'testapp':
+                print(ch)
+                chid = ch['id']
+    except Exception:
+        print(traceback.format_exc())
+    return chid
+
+def main():
+    spark = SparkSession.builder.appName("Slack").getOrCreate()
```
Comment (liororama) [inline components/Python/PySpark/slack-events/main.py]: Move to python

File: components/Python/PySpark/slack-events/main.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
+import os
+import json
+import argparse
+import traceback
+from datetime import datetime, timedelta
+from parallelm.mlops import mlops as pm
```
Comment (liororama) [inline components/Python/PySpark/slack-events/main.py]: Lets use mlops as the mlops prefix not pm

File: components/Python/PySpark/slack-events/main.py [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+            last_time = df.iloc[[-1]].value.values[0]
+        print("Last message retrieved at {} msec".format(last_time))
+    except Exception:
+        print(traceback.format_exc())
+        pass
+    
+    if last_time == -1:
+        evts = pm.get_events(is_alert=is_alert)
+    else:
+        #expects in sec, convert msec to sec
```
Comment (liororama) [inline components/Python/PySpark/slack-events/main.py]: Space after # in comment

File: components/Python/PySpark/slack-events/main.py [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+                               jsons['raiseAlert'],
+                               datetime.utcfromtimestamp(jsons['created']/1000),
+                               jsons['host'])
+        if jsons['eventType'] == 'ModelAccepted':
+            text_mesasge = text_message +\
+                            "Model ID: {}".format(jsons['modelId'])
+
+        ret = sc.api_call('chat.postMessage',channel=chid,text=text_message)
+
+    #TODO: Fix this bug - opaques dont work on spark
```
Comment (harshil2493) [inline components/Python/PySpark/slack-events/main.py]: open ticket?

**Review by harshil2493 — APPROVED** [review]

File: components/Python/PySpark/slack-events/main.py (hunk 4 above)
Comment (sriram109) [reply components/Python/PySpark/slack-events/main.py]: ill test it on python before updating here - until then ill leave it as pyspark

File: components/Python/PySpark/slack-events/main.py (hunk 7 above)
Comment (sriram109) [reply components/Python/PySpark/slack-events/main.py]: ref-5727 opened :)

**Comment by sriram109** [comment]
for anyone curious, this is how it looks on slack 
screen shot 2019-02-22 at 5 08 30 pm

**Review by harshilpm — APPROVED** [review]

File: components/Python/PySpark/slack-events/main.py (hunk 4 above)
Comment (sriram109) [reply components/Python/PySpark/slack-events/main.py]: switched to python (pr updated)

**Pull request merged**